source/common/upstream/upstream_impl.cc

```diff
@@ -1146,6 +1146,12 @@ void ClusterImplBase::setOutlierDetector(const Outlier::DetectorSharedPtr& outli
       [this](const HostSharedPtr& host) -> void { reloadHealthyHosts(host); });
 }
 
+void ClusterImplBase::setTransportFactoryContext(
+    std::unique_ptr<Server::Configuration::TransportSocketFactoryContext>
```

Comment:
Main nits remaining are to fix up these `unique_ptr` type names everywhere to follow the `FooPtr` style in https://github.com/envoyproxy/envoy/blob/main/STYLE.md#deviations-from-google-c-style-guidelines.